### PR TITLE
JDK-8265373: Change to GCC 10.3 for building on Linux at Oracle

### PR DESCRIPTION
**Pull request by erikj79**
Oracle is updating the version of GCC for building the JDK to 10.3.

In addition to the version bump, I'm also enabling cross compilation OOTB using jib from a linux-aarch64 to linux-x64 binaries, which was the last missing combination. (Native x64, native aarch64, cross from x64 to aarch64 all worked already and will continue to do so.)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265373](https://bugs.openjdk.java.net/browse/JDK-8265373): Change to GCC 10.3 for building on Linux at Oracle

### Reviewers
 * [Mikael Vidstedt](https://openjdk.java.net/census#mikael) (@vidmik - **Reviewer**) ⚠️ Review applies to dec3ae2ae26b6085f3679a1bb98e36b903480b6a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3551/head:pull/3551` \
`$ git checkout pull/3551`

Update a local copy of the PR: \
`$ git checkout pull/3551` \
`$ git pull https://git.openjdk.java.net/jdk pull/3551/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3551`

View PR using the GUI difftool: \
`$ git pr show -t 3551`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3551.diff">https://git.openjdk.java.net/jdk/pull/3551.diff</a>

</details>

### CODE REVIEW

File: make/conf/jib-profiles.js [hunk 1]
```diff
@@ -419,9 +419,11 @@ var getJibProfilesProfiles = function (input, common, data) {
         "linux-x64": {
             target_os: "linux",
             target_cpu: "x64",
-            dependencies: ["devkit", "gtest", "graphviz", "pandoc"],
-            configure_args: concat(common.configure_args_64bit,
+            dependencies: ["devkit", "gtest", "build_devkit", "graphviz", "pandoc"],
+            configure_args: concat(
+                (input.build_cpu == "x64" ? common.configure_args_64bit : "--openjdk-target=x86_64-linux-gnu"),
                 "--with-zlib=system", "--disable-dtrace",
+                (input.build_cpu != "x64" ? "--openjdk-target=x86_64-linux-gnu" : null),
```
Comment (vidmik) [inline make/conf/jib-profiles.js:426]: Doesn't this add `--openjdk-target=x86_64-linux-gnu` twice when build_cpu is not "x64"?

Comment (erikj79) [reply make/conf/jib-profiles.js]: You are correct, I never finished that edit. Uploaded fix.

**Review by vidmik — APPROVED** [review]

**Pull request closed without merging**